### PR TITLE
[android]  Make ExclusiveAppComponent accessible to FlutterFragment's subclasses similar to FlutterActivity's

### PR DESCRIPTION
**Pull request by 0xZOne**
In the `FlutterActivity` subclass, we can access the instance of `FlutterActivityAndFragmentDelegate` which is `protected`, as below show. 

```java
public class FlutterActivityEx extends FlutterActivity {
    private void performAttach() {
          // Attach plugins to the activity.
          // getFlutterEngine().getActivityControlSurface().attachToActivity(getActivity(), getLifecycle());
          getFlutterEngine().getActivityControlSurface().attachToActivity(delegate, getLifecycle());
    }
}
```

But, in `FlutterFragment`, the instance of `FlutterActivityAndFragmentDelegate` is package-level access.  so we expect to modify the `delegate` member to `protected`, similar to `FlutterActivity`. Make sure FlutterActivity/FlutterFragment subclasses can call `flutterEngine.getActivityControlSurface().attachToActivity(delegate, getLifecycle())`.

Related issues: [#issues/93905](https://github.com/flutter/flutter/issues/93905)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/android/FlutterFragment.java [hunk 1]
```diff
@@ -705,7 +705,7 @@ protected Bundle createArgs() {
   // Delegate that runs all lifecycle and OS hook logic that is common between
   // FlutterActivity and FlutterFragment. See the FlutterActivityAndFragmentDelegate
   // implementation for details about why it exists.
-  @VisibleForTesting @Nullable /* package */ FlutterActivityAndFragmentDelegate delegate;
+  @VisibleForTesting @Nullable protected FlutterActivityAndFragmentDelegate delegate;
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterFragment.java:708]: See comment https://github.com/flutter/flutter/issues/93905#issuecomment-989050277

I don't think this is the right fix

Comment (0xZOne) [reply shell/platform/android/io/flutter/embedding/android/FlutterFragment.java:708]: In the shared engine scenario (eg, [flutter_boost](https://github.com/alibaba/flutter_boost)), We expect to be able to take over ActivityControlSurface in the FlutterActivity/FlutterFragment subclass(eg, call `ActivityControlSurface.attachToActivity`), so here make ExclusiveAppComponent visibile to FlutterActivity/FlutterFragment subclass.

```java
  void attachToActivity(
      @NonNull ExclusiveAppComponent<Activity> exclusiveActivity, @NonNull Lifecycle lifecycle);
```

Comment (0xZOne) [reply shell/platform/android/io/flutter/embedding/android/FlutterFragment.java:708]: As stated in the comments https://github.com/flutter/flutter/issues/93905#issuecomment-989673591, We need this PR.

Comment (dnfield) [reply shell/platform/android/io/flutter/embedding/android/FlutterFragment.java:708]: Can we instead expose the method we actually need to subclasses.

Comment (0xZOne) [reply shell/platform/android/io/flutter/embedding/android/FlutterFragment.java:708]: > Can we instead expose the method we actually need to subclasses.

Thank you for your reply.

In this case, I think exposing the `delegate` of `FlutterFragment` as a `protected` member to subclasses is the least costly solution.

**Comment by DavidMGT** [comment]
请尽快推进吧我们升级了 最新的flutter版本无法使用了

**Comment by gaoyangcr7** [comment]
关注

**Comment by Duane030414** [comment]
mark

**Comment by houziruinb87** [comment]
same issue

**Comment by chinmaygarde** [comment]
It seems like @blasten and @0xZOne are in [disagreement about this approach](https://github.com/flutter/flutter/issues/93905#issuecomment-989050277) and @dnfield has a [suggested workaround](https://github.com/flutter/engine/pull/30126#discussion_r765875835). Since there is not consensus among reviewers about this being the right approach, I don't think progress here is possible. Closing.

**Comment by 0xZOne** [comment]
> It seems like @blasten and @0xZOne are in [disagreement about this approach](https://github.com/flutter/flutter/issues/93905#issuecomment-989050277) and @dnfield has a [suggested workaround](https://github.com/flutter/engine/pull/30126#discussion_r765875835). Since there is not consensus among reviewers about this being the right approach, I don't think progress here is possible. Closing.

I admit @dnfield's suggested workaround is more elegant. ;)  I have reworked.

**Pull request closed without merging**